app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -0,0 +1,147 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sagebionetworks.bridge.models.CriteriaUtils.SPECIFICITY_SORTER;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Resource;
+
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.SubpopulationDao;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.CriteriaUtils;
+import org.sagebionetworks.bridge.models.schedules.ScheduleContext;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifierImpl;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.FailedBatch;
+import com.google.common.collect.Lists;
+
+@Component
+public class DynamoSubpopulationDao implements SubpopulationDao {
+    
+    private DynamoDBMapper mapper;
+
+    /** DynamoDB mapper for the HealthDataRecord table. This is configured by Spring. */
+    @Resource(name = "subpopulationDdbMapper")
+    public void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Override
+    public Subpopulation createSubpopulation(Subpopulation subpop) {
+        checkNotNull(subpop);
+        checkNotNull(subpop.getStudyIdentifier());
+
+        subpop.setVersion(null);
+        subpop.setDeleted(false);
+        subpop.setGuid(BridgeUtils.generateGuid());
+        mapper.save(subpop);
+        return subpop;
```

Comment:
Minor: As written, this will return the subpop that was submitted to DDB, but not the subpop that was saved. (Specifically, the version attribute will be bumped, and anything with DynamoDBIgnore will be removed.)

This might not be a big deal, but if a client creates a subpop, edits the returned JSON, and then immediately re-submits it, they'll get a ConcurrentModificationException because of the version number mismatch. The way we handle it in Health Data Records is to just return the guid.

This also affects update() and createDefault()